include/swift/SIL/OwnershipUtils.h

```diff
@@ -531,6 +533,22 @@ class InteriorPointerOperandKind {
     }
   }
 
+  /// Given a \p value that is a result of an instruction with an interior
+  /// pointer operand, return the interior pointer operand kind that would be
+  /// appropriate for that operand.
+  static Optional<InteriorPointerOperandKind> inferFromResult(SILValue value) {
```

Comment:
The `Optional` is just an extra layer of abstraction and overhead that isn't more helpful than discriminating Invalid cases in the enum.

Reply:
It is useful in the sense that it documents in the code that one needs to handle failure there and prevents you from not handling it.

Reply:
Ok. I will change this.